packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -185,6 +199,80 @@ def _ai_env_status_for_provider_text(self, provider_id):
         
         return output + "\n"
 
+    # Is this a name of a Python variable that can be called as a LangChain chain?
+    def _is_langchain_chain(self, name):
+        # Reserved word in Python?
+        if (keyword.iskeyword(name)):
+            return False;
+    
+        acceptable_name = re.compile('^[a-zA-Z0-9_]+$')
+        if (not acceptable_name.match(name)):
+            return False;
+        
+        ipython = get_ipython()
+        return(name in ipython.user_ns and isinstance(ipython.user_ns[name], LLMChain))
+
+    # Is this an acceptable name for an alias?
+    def _validate_name(self, register_name):
+        # A registry name contains ASCII letters, numbers, hyphens, underscores,
+        # and periods. No other characters, including a colon, are permitted
+        acceptable_name = re.compile('^[a-zA-Z0-9._-]+$')
+        if (not acceptable_name.match(register_name)):
+            raise ValueError('A registry name may contain ASCII letters, numbers, hyphens, underscores, '
+                + 'and periods. No other characters, including a colon, are permitted')
+
+    # Initially set or update an alias to a target
+    def _safely_set_target(self, register_name, target):
+        # If target is a string, treat this as an alias to another model.
+        if self._is_langchain_chain(target):
+            ip = get_ipython()
+            self.custom_model_registry[register_name] = ip.user_ns[target]
+        else:
+            # Ensure that the destination is properly formatted
+            if (':' not in target):
+                raise ValueError(
+                    'Target model must be an LLMChain object or a model name in PROVIDER_ID:MODEL_NAME format')
+
+            self.custom_model_registry[register_name] = target
+
+    def handle_delete(self, args: DeleteArgs):
+        if (args.name in AI_COMMANDS):
+            raise ValueError(f"Reserved command names, including {args.name}, cannot be deleted")
+        
+        if (args.name not in self.custom_model_registry):
+            raise ValueError(f"There is no alias called {args.name}")
+        
+        del self.custom_model_registry[args.name]
+        output = f"Deleted alias `{args.name}`"
+        return TextOrMarkdown(output, output)
+
+    def handle_register(self, args: RegisterArgs):
+        # Existing command names are not allowed
+        if (args.name in AI_COMMANDS):
+            raise ValueError(f"The name {args.name} is reserved for a command")
+        
+        # Existing registered names are not allowed
+        if (args.name in self.custom_model_registry):
+            raise ValueError(f"The name {args.name} is already associated with a custom model; "
```

Comment:
These errors currently produce a full stack trace which is not relevant to the user. See the attached screenshot. Is there a way to show the error without spitting the full stack trace?
<img width="852" alt="Screen Shot 2023-06-06 at 12 26 49 PM" src="https://github.com/jupyterlab/jupyter-ai/assets/289369/2dc873cf-0133-47d5-bc72-0650c1e84ea5">

Reply:
I just pushed a change that wraps `ValueError` exceptions so that they print to `stderr` without this full stack trace, which I agree is distracting and unnecessary

<img width="733" alt="image" src="https://github.com/jupyterlab/jupyter-ai/assets/93281816/5ec2635e-19e3-4d87-8d9b-32bc895cc2e0">